### ISSUE TITLE
useClickOutside seems to prevent `this` from resolving controller in a disconnect

### ISSUE THREAD

**Issue by mepatterson** [stimulus-use/stimulus-use#36]
#### Without useClickOutside (expected behavior)
```
import { Controller } from "stimulus";
import { useClickOutside } from "stimulus-use";

export default class extends Controller {
  initialize() {}

  connect() {
    super.connect();

    console.log(this);
    console.log(this);
  }

  disconnect() {
    console.log(this);
  }
}
```
![Screen Shot 2020-08-26 at 1 21 11 PM](https://user-images.githubusercontent.com/74207/91341439-241e4000-e79f-11ea-97c4-e6db93a76683.png)
(the first 2 log messages are the two from connect, the 3rd one is when you navigate to another page and it disconnects)

#### With useClickOutside (unexpected behavior)
```
import { Controller } from "stimulus";
import { useClickOutside } from "stimulus-use";

export default class extends Controller {
  initialize() {}

  connect() {
    super.connect();

    console.log(this);
    useClickOutside(this);
    console.log(this);
  }

  disconnect() {
    console.log(this);
  }
}
```
![Screen Shot 2020-08-26 at 1 23 21 PM](https://user-images.githubusercontent.com/74207/91341611-592a9280-e79f-11ea-8ce2-ae87b53fecd3.png)
(notice the 3rd one is undefined when you navigate away and it runs the disconnect)

Is there something I'm doing wrong? Or should I not be expecting to have `this` resolve to the controller in a disconnect after using this?

**Comment by adrienpoly** [comment]
Ok I think you pointed a valid issue here. Thanks for reporting it very clearly. I ll try to submit a fix quickly

**Comment by adrienpoly** [comment]
I have just released 0.12.0 https://github.com/stimulus-use/stimulus-use/releases/tag/v0.12.0

it should fix this issue 

On a side note, for the connect method _"normally"_ you don't need to call super

**Comment by adrienpoly** [comment]
@all-contributors add @mepatterson to bug
